lib/lightning/projects.ex

```diff
@@ -163,7 +163,12 @@ defmodule Lightning.Projects do
     jobs =
       list_projects_having_history_retention()
       |> Enum.map(fn project ->
-        new(%{project_id: project.id, type: "data_retention"}, max_attempts: 3)
+        delay = :rand.uniform(7_200) - 1
```

Comment:
Thinking only about the timing, there are other jobs that run between 2am and 3am so what about anticipating it to 30 1 * * *? 
This would prevent an eventual collision other cron jobs that come after this one from 1 2 * * * 

```
base_cron = [
      ...
      {"1 2 * * *", Lightning.Projects, args: %{"type" => "data_retention"}},
      ...
    ]
```
```
    cleanup_cron =
      if Application.get_env(:lightning, :purge_deleted_after_days) > 0,
        do: [
          {"15 2 * * *", Lightning.Accounts, args: %{"type" => "purge_deleted"}},
          {"30 2 * * *", Lightning.Credentials,
           args: %{"type" => "purge_deleted"}},
          {"45 2 * * *", Lightning.Projects, args: %{"type" => "purge_deleted"}},
       ],
        else: []
```